src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs

```diff
@@ -0,0 +1,347 @@
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.InlineDeclaration
+{
+    public partial class CSharpInlineDeclarationTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return new Tuple<DiagnosticAnalyzer, CodeFixProvider>(
+                new CSharpInlineDeclarationDiagnosticAnalyzer(),
+                new CSharpInlineDeclarationCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariable1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (int.TryParse(v, out int i))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInCSharp6()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}", parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp6));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariablePreferVar1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(string v)
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M(string v)
+    {
+        if (int.TryParse(v, out var i))
+        {
+        } 
+    }
+}", options: UseImplicitTypeTests.ImplicitTypeEverywhere());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariablePreferVarExceptForPredefinedTypes1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(string v)
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M(string v)
+    {
+        if (int.TryParse(v, out int i))
+        {
+        } 
+    }
+}", options: UseImplicitTypeTests.ImplicitTypeButKeepIntrinsics());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestAvailableWhenWrittenAfter1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        }
+        i = 0;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (int.TryParse(v, out int i))
+        {
+        }
+        i = 0;
+    }
+}");
+        }
```

Comment:
Should we also have tests where the same variable is used as an out parameter in multiple invocations?